comp/core/clusterchecks/def/component.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package clusterchecks provides the cluster checks handler component
+package clusterchecks
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
+)
+
+// team: container-platform
+
+// Component is the interface for the cluster checks handler
+type Component interface {
+	// Run is the main goroutine for the handler. It has to
+	// be called in a goroutine with a cancellable context.
+	Run(ctx context.Context)
+
+	// API Methods
+
+	// RejectOrForwardLeaderQuery rejects or forwards the query based on leadership status
+	RejectOrForwardLeaderQuery(rw http.ResponseWriter, req *http.Request) bool
```

Comment:
I think this is an implementation detail that shouldn't be exposed in the interface.
I think this is only used by functions defined below, so we should refactor to avoid exposing this.